### PR TITLE
C#: `AttributeCollection` is no longer considered a HTML sink.

### PR DESCRIPTION
**Pull request by michaelnebel**
In this PR we remove the indexer and `Add` method of `System.Web.UI.AttributeCollection` as possible HTML sinks as the information is HTML encoded when the control is rendered.

This can be confirmed by running (needs to be a .NET Framework app).
```csharp
using System;
using System.Web.UI;
using System.Web.UI.WebControls;

class Program
{
    static void Main(string[] args)
    {
        Button b = new Button();
        b.Attributes.Add("my_key", "<some value & my stuff");
        b.Attributes["my_index_key"] = "my_<index&_value";
        b.RenderControl(new HtmlTextWriter(Console.Out));
    }
}
```

### CODE REVIEW

File: csharp/ql/lib/semmle/code/csharp/security/dataflow/flowsinks/Html.qll [hunk 1]
```diff
@@ -56,13 +56,11 @@ class HtmlTextWriterSink extends HtmlSink {
...
 /**
- * An expression that is used as an argument to an HTML sink method on
- * `AttributeCollection`.
+ * DEPRECATED: Attribute collections are no longer considered HTML sinks.
  */
-class AttributeCollectionSink extends HtmlSink {
+deprecated class AttributeCollectionSink extends DataFlow::ExprNode {
   AttributeCollectionSink() {
     exists(SystemWebUIAttributeCollectionClass ac, Parameter p |
-      p = ac.getAddMethod().getParameter(1) or
```
Comment (hvitved) [inline csharp/ql/lib/semmle/code/csharp/security/dataflow/flowsinks/Html.qll:65]: Why is this removed?

Comment (michaelnebel) [reply csharp/ql/lib/semmle/code/csharp/security/dataflow/flowsinks/Html.qll:65]: Bah, stupid mistake (first just removing it for testing and then deprecated the stuff afterwards). Thank you for catching.

**Review by hvitved — APPROVED** [review]

**Comment by michaelnebel** [comment]
DCA looks good!
